pages/blog/getting-the-best-out-of-your-graphql-client-by-using-relay-patterns-through-graphql-code-generator.mdx

```diff
@@ -0,0 +1,212 @@
+  fragment FriendList_QueryFragment on Query {
+    friends(first: 5) {
+      id
+      ...UserListItem_UserFragment
+    }
+  }
+`)
+
+type FriendListProps = {
+  query: FragmentType<typeof FriendList_QueryFragment>
+}
+
+export function FriendList(props: FriendListProps) {
+  const query = useFragment(FriendList_QueryFragment, props.query)
+  return (
+    <List>
+      {query.friends.map((user) => (
+        <FriendListItem key={user.id} user={user} />
+      ))}
+    </List>
+  )
+}
+```
+
+Earlier, we declared the `Avatar` component with the `Avatar_UserFragment` fragment. We can now re-use this Avatar in a different context by again spreading the fragment and then passing the relevant data to the `Avatar` component `user` property.
+
+```tsx
+import { gql, useFragment, FragmentType } from '@gql'
+
+const UserProfileHeader_QueryFragment = gql(/* GraphQL */ `
+  fragment UserProfileHeader_QueryFragment on Query {
+    viewer {
+      id
+      homeTown
+      registeredAt
+      ...Avatar_UserFragment
+    }
+  }
+`)
+
+type UserProfileHeaderProps = {
+  query: FragmentType<typeof UserProfileHeader_QueryFragment>
+}
+
+export function UserProfileHeader(props: UserProfileHeaderProps) {
+  const query = useFragment(UserProfileHeader_QueryFragment, props.query)
+  return (
+    <>
+      <Avatar user={query.viewer} />
+      {query.viewer.homeTown}
+      {query.viewer.registeredAt}
+    </>
+  )
+}
+```
+
+Now, we have two UI components that require data from the root Query type, `FriendList` and `UserProfileHeader` .
+
+## Compose all Query fragments into a single Query operation
+
+Finally, we can spread all our fragments into a single GraphQL Query operation that fetches all the data on our route component. This allows us to efficiently fetch all the data required for the route in one server roundtrip.
+
+```jsx
+import { gql, useFragment, FragmentType } from '@gql'
+import { UserProfileHeader } from './user-profile-header'
+import { FriendList } from './friend-list'
+
+const UserProfileRoute_Query = gql(/* GraphQL */ `
+  query UserProfile_Query {
+    ...UserProfileHeader_QueryFragment
+    ...UserList_QueryFragment
+  }
+`)
+
+export function UserProfileRoute_Query() {
+  const { data, loading, error } = useQuery(UserProfile_Query)
+  if (loading) return <Loading />
+  if (error) return <Error />
+  return (
+    <>
+      <FriendList query={data} />
+      <UserProfileHeader query={data} />
+    </>
+  )
+}
+```
+
+## Conclusion
+
+This pattern allows you to re-use and scale your components to the maximum while having a clear and sane data dependency flow from the top route down to the ends of the UI component trees.
+
+You can start adopting this pattern within your GraphQL application with any GraphQL client that supports `TypedDocumentNode` through our new and battle-tested GraphQL Code Generator preset `[gql-tag-operations-preset](https://www.graphql-code-generator.com/plugins/presets/gql-tag-operations-preset)`.
```

Comment:
let’s put a short list of the compatible frameworks here?
ex: Apollo client, Apollo Angular, URQL and Vue

Reply:
yes and let's mention that we are working on react-query and graphql-request (and later go back and link to our PRs there)